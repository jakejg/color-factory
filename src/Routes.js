import React, { useState } from 'react';
import { Route, Switch, Redirect } from 'react-router-dom';
import FilterColors from './FilterColors';
import ColorList from './ColorList';
import AddColorForm from './AddColorForm'
import useLocalStorage from './useLocalStorage'

function Routes() {
    const INITIAL_STATE = [
        {name:"Red", hexCode: '#FF0000'},
        {name:"Green", hexCode: '#008000'},
        {name:"Aqua", hexCode: '#00FFFF'}
    ]
    const [colors, setColors ] = useLocalStorage('colors', INITIAL_STATE)

    return (
        <div>
            <Switch>
                <Route exact path='/colors'>   
                    <ColorList colors={colors}/>
                </Route>
                <Route exact path='/colors/new'>
                    <AddColorForm addColor={setColors}/>
                </Route>
                <Route exact path='/colors/:color'>
                    <FilterColors colors={colors} />
                </Route>
                <Route>
                    <Redirect to='/colors' />
                </Route>
            </Switch>
        </div>
    );
}

export default Routes;



