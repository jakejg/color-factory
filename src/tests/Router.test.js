import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import Routes from '../Routes';

test('renders without crashing', () => {
  render(
<MemoryRouter initialEntries={['/colors']} >
    <Routes />
</MemoryRouter>)
});

test('matches snapshot', () => {
    const { asFragment } = render(
    <MemoryRouter initialEntries={['/colors']} >
        <Routes />
    </MemoryRouter>);
    expect( asFragment()).toMatchSnapshot();
  });

test('displays home page', () => {
    const { queryByText } = render(
        <MemoryRouter initialEntries={['/colors']} >
            <Routes />
        </MemoryRouter>);
    expect(queryByText('Select a Color')).toBeInTheDocument();
})

test('displays a color page', () => {
    const { queryByText } = render(
        <MemoryRouter initialEntries={['/colors/red']} >
            <Routes />
        </MemoryRouter>);
    expect(queryByText('This is Red!')).toBeInTheDocument();
})

test('renders home on a bad route', () => {
    const { queryByText } = render(
        <MemoryRouter initialEntries={["/bad"]}>
          <Routes />
        </MemoryRouter>);
    expect(queryByText('Select a Color')).toBeInTheDocument();
})

test('renders home on a color that doesnt exist', () => {
    const { queryByText } = render(
        <MemoryRouter initialEntries={["/colors/bluebeerry"]}>
          <Routes />
        </MemoryRouter>);
    expect(queryByText('Select a Color')).toBeInTheDocument();
})

test('can add a color', () => {
    const { queryByText, queryByLabelText } = render(
        <MemoryRouter initialEntries={["/colors/new"]}>
          <Routes />
        </MemoryRouter>);
    const name = queryByLabelText('Choose a name for your color');
    const hexCode = queryByLabelText('Choose color')
    fireEvent.change(name, {target: {value: 'Purple'}})
    fireEvent.change(hexCode, {target: {value: '#800080'}})
    fireEvent.click(queryByText('Select'));
    
    expect(queryByText('Purple')).toBeInTheDocument();
})