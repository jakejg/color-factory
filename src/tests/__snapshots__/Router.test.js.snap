// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="Colors"
    >
      <h1
        class="Colors-title"
      >
        Select a Color
      </h1>
      <ul
        class="Colors-list"
      >
        <li>
          <a
            href="/colors/Red"
          >
            Red
          </a>
        </li>
        <li>
          <a
            href="/colors/Green"
          >
            Green
          </a>
        </li>
        <li>
          <a
            href="/colors/Aqua"
          >
            Aqua
          </a>
        </li>
      </ul>
      <h2>
        Or add a 
        <a
          href="/colors/new"
        >
          color
        </a>
      </h2>
    </div>
  </div>
</DocumentFragment>
`;
